### ISSUE TITLE
'gatk' and 'picard' module broke

### ISSUE THREAD

**Issue by Ofsm** [ewels/MultiQC#1027]
Hello, I'm trying to run the `multiqc .` and I'm getting these errors with gatk and picard modules:

```console
oscar@oscarpc:~$ multiqc .
/home/oscar/anaconda2/lib/python2.7/site-packages/multiqc/utils/config.py:45: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  configs = yaml.load(f)
/home/oscar/anaconda2/lib/python2.7/site-packages/multiqc/utils/config.py:51: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  sp = yaml.load(f)
[INFO   ]         multiqc : This is MultiQC v1.7
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching '.'
Searching 170638 files..  [####################################]  100%          
[ERROR  ]         multiqc : Oops! The 'gatk' MultiQC module broke... 
  Please copy the following traceback and report it at https://github.com/ewels/MultiQC/issues 
  If possible, please include a log file that triggers the error - the last file found was:
    ./MultiQC/multiqc/modules/gatk/varianteval.py
============================================================
Module gatk raised an exception: Traceback (most recent call last):
  File "/home/oscar/anaconda2/bin/multiqc", line 440, in multiqc
    output = mod()
  File "/home/oscar/anaconda2/lib/python2.7/site-packages/multiqc/modules/gatk/gatk.py", line 41, in __init__
    n_reports_found += self.parse_gatk_varianteval()
  File "/home/oscar/anaconda2/lib/python2.7/site-packages/multiqc/modules/gatk/varianteval.py", line 20, in parse_gatk_varianteval
    parsed_data = parse_single_report(f['f'])
  File "/home/oscar/anaconda2/lib/python2.7/site-packages/multiqc/modules/gatk/varianteval.py", line 106, in parse_single_report
    d[headers[i]] = s
IndexError: list index out of range
============================================================
[ERROR  ]         multiqc : Oops! The 'picard' MultiQC module broke... 
  Please copy the following traceback and report it at https://github.com/ewels/MultiQC/issues 
  If possible, please include a log file that triggers the error - the last file found was:
    ./anaconda2/lib/python2.7/site-packages/multiqc/modules/picard/ValidateSamFile.pyc
============================================================
Module picard raised an exception: Traceback (most recent call last):
  File "/home/oscar/anaconda2/bin/multiqc", line 440, in multiqc
    output = mod()
  File "/home/oscar/anaconda2/lib/python2.7/site-packages/multiqc/modules/picard/picard.py", line 95, in __init__
    n['ValidateSamFile'] = ValidateSamFile.parse_reports(self)
  File "/home/oscar/anaconda2/lib/python2.7/site-packages/multiqc/modules/picard/ValidateSamFile.py", line 91, in parse_reports
    data = _parse_reports_by_type(self)
  File "/home/oscar/anaconda2/lib/python2.7/site-packages/multiqc/modules/picard/ValidateSamFile.py", line 125, in _parse_reports_by_type
    first_line = filehandle.readline().rstrip()
  File "/home/oscar/anaconda2/lib/python2.7/codecs.py", line 314, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xf3 in position 1: invalid continuation byte
============================================================
[INFO   ]        samtools : Found 1 rmdup reports
[INFO   ]         bowtie2 : Found 1 reports
[WARNING]        bargraph : Tried to make bar plot, but had no data
[INFO   ]         multiqc : Compressing plot data
[WARNING]         multiqc : Previous MultiQC output found! Adjusting filenames..
[WARNING]         multiqc : Use -f or --force to overwrite existing reports instead
[INFO   ]         multiqc : Report      : multiqc_report_1.html
[INFO   ]         multiqc : Data        : multiqc_data_1
[INFO   ]         multiqc : MultiQC complete
```

I run this :
 - Command used to run MultiQC: `multiqc .`
 - MultiQC Version: MultiQC v1.7
 - Operating System: Ubuntu 18.04
 - Python Version: Python 2.7.16
 - Method of MultiQC installation: conda

THANKS FOR THE HELP

**Comment by ewels** [comment]
Hi @Ofsm,

Please could you attach some files that trigger these errors? Then I can replicate the problem at my end and look into fixing it.

Thanks,

Phil

**Comment by Ofsm** [comment]
Hello E, well I don't know if I'm doing this in the right way, I just use the command 'multiqc .' in my Home terminal and that was the result I got.

Do i need to specify the directory where I have my files?

**Comment by ewels** [comment]
> Do i need to specify the directory where I have my files?

The `.` after the `multiqc` specifies the current working directory (same as `./`), so you're already doing that 😄 

MultiQC is finding your files in that directory but breaking. Ohhhh, wait. When I read your logs in a bit more detail I think I see the problem:

```
  If possible, please include a log file that triggers the error - the last file found was:
    ./MultiQC/multiqc/modules/gatk/varianteval.py
```
```
  If possible, please include a log file that triggers the error - the last file found was:
    ./anaconda2/lib/python2.7/site-packages/multiqc/modules/picard/ValidateSamFile.pyc
```

Ok, so I think that you're probably doing some weird stuff here 😆 It looks like MultiQC is attempting to create a report from its own source code. Which makes no sense..

1. You appear to have downloaded the MultiQC source code (`./MultiQC/multiqc/`) and also installed using bioconda (`./anaconda2/lib/python2.7/site-packages/multiqc/`). If you've installed using conda, you don't need the source files. See https://multiqc.info/docs/#installing-multiqc
2. You should probably read the usage instructions for MultiQC 😉 https://multiqc.info/docs/#running-multiqc
    * In short, you need to pass it the paths to some directories that have results / lot files from your analysis. MultiQC does not run any tools for you, it only parses the outputs from tools into a report.

Ok, I hope that helps! I'll close this issue now, but feel free to add more comments if you need any help.

Phil
